stdlib/LinearAlgebra/test/generic.jl

```diff
@@ -248,6 +253,31 @@ end
     end
 end
 
+@testset "normalize for multidimensional arrays" begin
+    function test_arr(arr)
+        @test normalize(arr) == normalize!(copy(arr))
+        @test size(normalize(arr)) == size(arr)
+        @test vec(normalize(arr)) == normalize(vec(arr))
+    end
+
+    arr = [ 1.0 0.0; 0.0 1.0 ]
+    test_arr(arr)
+
+    arr = [
+            [1.0 0.0 0.0];
+            [0.0 1.0 0.0]
+         ]
+    test_arr(arr)
+
+    arr = randn(2,3)
+    test_arr(arr)
```

Comment:
I don't really see the point of having three test cases for 2d arrays here.  Seems like
```
test_arr([1 2 3; 4 5 6])
test_arr(OffsetArray([-1,0], (-2,)))
```
would be sufficient to test all of the cases.

In such situations, rather than define a `test_arr` function, it is more idiomatic to use a loop:
```
for A in ([1 2 3; 4 5 6], OffsetArray([-1,0], (-2,)))
    @test normalize(arr) == normalize!(copy(arr))
    @test size(normalize(arr)) == size(arr)
    @test vec(normalize(arr)) == normalize(vec(arr))
end
```

Reply:
I removed the redundant test cases in cdb2167.

An interesting thing popped up:

```
 using LinearAlgebra;
 
arr = [ 1 2 3; 4 5 6 ] 
normalize!(copy(arr))
```
Actually throws an error somewhere in the trace stack its trying to do a Int64 cast of a decimal:

```
$ ./julia foo.jl 
ERROR: LoadError: InexactError: Int64(0.10482848367219183)
Stacktrace:
 [1] Int64 at ./float.jl:710 [inlined]
 [2] convert at ./number.jl:7 [inlined]
 [3] setindex! at ./array.jl:825 [inlined]
 [4] macro expansion at /Users/ssikdar1/julia/usr/share/julia/stdlib/v1.5/LinearAlgebra/src/generic.jl:150 [inlined]
 [5] macro expansion at ./simdloop.jl:77 [inlined]
 [6] rmul!(::Array{Int64,2}, ::Float64) at /Users/ssikdar1/julia/usr/share/julia/stdlib/v1.5/LinearAlgebra/src/generic.jl:149
 [7] __normalize! at /Users/ssikdar1/julia/usr/share/julia/stdlib/v1.5/LinearAlgebra/src/generic.jl:1602 [inlined]
 [8] normalize!(::Array{Int64,2}, ::Int64) at /Users/ssikdar1/julia/usr/share/julia/stdlib/v1.5/LinearAlgebra/src/generic.jl:1593
 [9] normalize!(::Array{Int64,2}) at /Users/ssikdar1/julia/usr/share/julia/stdlib/v1.5/LinearAlgebra/src/generic.jl:1592
 [10] top-level scope at /Users/ssikdar1/julia/foo.jl:5
 [11] include(::Module, ::String) at ./Base.jl:377
 [12] exec_options(::Base.JLOptions) at ./client.jl:288
 [13] _start() at ./client.jl:484
in expression starting at /Users/ssikdar1/julia/foo.jl:5
``` 

When the array is changed to float numbers:
```
using LinearAlgebra;
   
arr = [ 1.0 2.0 3.0; 4.0 5.0 6.0 ]
normalize!(copy(arr))
```
the code works.

Should the cast issue be resolved in this PR or should I open a new github issue?

Reply:
If you run `@show typeof([ 1 2 3; 4 5 6 ] )` you will see it is an array of `Int`. 
`normalize!` tries to mutate the array inplace. This cannot change the eltype of the array, it will still be an array of `Int`. But it is very hard for an array of `Int` to have norm one (it means there is a single nonzero entry that is +-1). So I would say it is expected that `normalize!` almost always throws an error on `Int` arrays. However `normalize` without `!` should work on `Int` arrays and return `Float64` arrays. This should be tested.